src/directory/poll_watcher.rs

```diff
@@ -0,0 +1,73 @@
+use crate::core::META_FILEPATH;
+use crate::directory::error::OpenDirectoryError;
+use crate::directory::{WatchCallback, WatchCallbackList, WatchHandle};
+use std::io;
+use std::path::Path;
+use std::sync::Arc;
+use std::thread;
+use std::time::{Duration, UNIX_EPOCH};
+use std::fs;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+
+const POLL_INTERVAL_MS: u64 = 10;
```

Comment:
`500ms` would be a more sensible default I believe.

Reply:
can we make it configurable? that would be nice espeically for unit tests.

Reply:
I will try to make it configureable.